File: ruby/request.rb
require 'httparty'
require 'json'

base_url = 'https://cloud.mongodb.com/api/atlas/v1.0/'
options = {
  :digest_auth => {
    :username=>ENV['ATLAS_USER'],
    :password=>ENV['ATLAS_USER_KEY']
  }
}

result = HTTParty.get(base_url, options)

pp JSON.parse(result.body())

File: node/request.js
const urllib = require('urllib');

const baseUrl = 'https://cloud.mongodb.com/api/atlas/v1.0/';
const { ATLAS_USER, ATLAS_USER_KEY } = process.env;
const options = {
  digestAuth: `${ATLAS_USER}:${ATLAS_USER_KEY}`,
};

urllib.request(baseUrl, options, (error, data, response) => {
  if (error || response.statusCode !== 200) {
    console.error(`Error: ${error}`);
    console.error(`Status code: ${response.statusCode}`);
  } else {
    console.log(JSON.parse(data));
  }
});
